EC2: Allow tags with value None

eyj opened the pull request:
Currrently, EC2 tags with no value are not properly set, as it triggers logic that thinks the tag has not been set correctly. This PR adds an explicit check for this case.

eyj committed:
Allow setting of tags with value None

rallytime commented:
I believe setting tags to `None` was an intended behavior to handle deleting tags. However, you have a very valid point. I think we need to find a way to handle both use cases before merging this. Perhaps a `delete_tag` option, with lots of warnings in the docs about the behavior change? If that option is implemented, the proper [deprecation paths](https://docs.saltstack.com/en/latest/topics/development/deprecations.html) will need to be implemented so we don't break anyone's current behavior. What do you think @eyj? Do you have another idea of how to handle both use cases?

eyj commented:
Hmm. I think I only modified code that checks for successful execution; and removes one 'false positive'. AFAIK the current behaviour for a tag with None value is already to set it to a tag with None value, and then fail because the (now modified) code thinks it has not been set.

Tag deletion can anyway not be done with the CreateTags API call, there is a del_tags function further down. So I think this is fine :)

Do you think the test failure is due to something I changed? It does not look related to me.

rallytime commented:
Ah, ok. I see your point. Thanks for pointing that out. I didn't realize there was a different function for del_tags when I was looking at your changes. 

The test failures are unrelated. :)
